cnf-certification-test/networking/suite.go

```diff
@@ -153,9 +153,11 @@ func testUndeclaredContainerPortsUsage(env *provider.TestEnvironment) {
 
 		// Verify that all the listening ports have been declared in the container spec
 		failedPod := false
+		var listeningPortsAlreadyReported []netutil.PortInfo // short circuit to not report the same port more than once
 		for listeningPort := range listeningPorts {
-			if !declaredPorts[listeningPort] {
-				tnf.ClaimFilePrintf("%s is listening on port %d protocol %d, but that port was not declared in any container spec.", put, portInfo.PortNumber, portInfo.Protocol)
+			if !declaredPorts[listeningPort] && !netcommons.ExistsInPortInfoSlice(listeningPort, listeningPortsAlreadyReported) {
```

Comment:
I think this filter you add here is hiding a possible bug. I've reviewed the code of this TC (testUndeclaredContainerPortsUsage) but I still don't see why it should print the same port several times... It doesn't make sense. This filter shouldn't be necessary. We're missing something here.

This tc gets the list of declared ports in the containers manifest of each pod and then gets the list of all ports that are actually opened (and listening) by the containers' apps on that pod only. As all containers in a pod share the same networking namespace, it's not possible for the same port to be opened (and listening) more than 1 time. That's why we run the `ss` command through `nsenter` on the first container of that pod only: we should also see the ports opened by the other containers in the same pod.

Reply:
@greyerof I was just playing around with the `ss` command on my kind cluster and I see:
```
[root@kind-worker3 /]# ss -tulwn | grep 9100
tcp   LISTEN 0      4096       127.0.0.1:9100       0.0.0.0:*          
tcp   LISTEN 0      4096      172.18.0.4:9100       0.0.0.0:*    
```

I wonder if that's what's happening with the multiple printouts where the localhost and the IP of the host are showing up in the list twice (or more).

In my case above `172.18.0.4` is the IP of my worker3.

Reply:
Yep. I don't think that worker's port appears when using nsenter, so there should be something else.

Reply:
I will update my PR to remove this part of the cleanup so we can focus on a potential fix to the `ss` command or however we interpret the output.